client/src/components/Navbar.js

```diff
@@ -38,11 +40,25 @@ export default class Navbar extends Component {
             )
         })
     }
+
+    handleOffsetChange() {
+        window.location.reload()
```

Comment:
I would have liked to avoid refresh, but for one reason or another it would simply not return the updated userDetails prop in time to the component - this meant that if a user changed their timezone, it wouldn't always update the component they were viewing

Since this will usually be a onetime fire and forget feature, where it hopefully only needs to be set once, I don't think a reload is a big problem

Reply:
Agreed. I'm on board.